core/src/loader.rs

```diff
@@ -1584,7 +1585,92 @@ impl<'gc> Loader<'gc> {
                             &mut activation.context,
                             "complete",
                         );
-                        Avm2::dispatch_event(uc, complete_evt, sound_object);
+                        Avm2::dispatch_event(&mut activation.context, complete_evt, sound_object);
+
+                        let id3 = activation
+                            .avm2()
+                            .classes()
+                            .id3info
+                            .construct(&mut activation, &[])
+                            .expect("failed to construct ID3Info object");
+                        let tag = Tag::read_from(body.as_slice());
+                        if let Ok(ref tag) = tag {
+                            if let Some(v) = tag.album() {
+                                id3.set_public_property(
+                                    "album",
+                                    Avm2Value::String(AvmString::new_utf8(activation.gc(), v)),
```

Comment:
Does `AvmString::new_utf8(activation.gc(), v).into()` suffice here (and for all the other fields, below)?